drivers/mge-hid.c

```diff
@@ -1151,6 +1151,142 @@ static info_lkp_t eaton_input_bypass_mode_off_info[] = {
+	return eco_switchable_str;
+}
+
+/* Conversion back of eaton_input_eco_mode_auto_on_fun() */
+static double eaton_input_eco_mode_auto_on_nuf(const char *value)
+{
+	const char *bypass_switch_on_str = NULL;
+	const char *eco_switchable_str = NULL;
+	double eco_mode_auto_on = 1;
+	double eco_mode_auto_off = 0;
+
+	NUT_UNUSED_VARIABLE(value);
+
+	/* Check if input.bypass.switch.on is disabled and set it to 'on' */
+	bypass_switch_on_str = dstate_getinfo("input.bypass.switch.on");
+	if (!strcmp(bypass_switch_on_str, "disabled")) {
+		bypass_switch_on_str = eaton_input_bypass_check_range(1);
+	} else {
+		upsdebugx(1, "%s: Bypass switch on state is: %s , must be disabled before switching on", __func__, bypass_switch_on_str);
+		return eco_mode_auto_off;
+	}
+
+	/* Check if input.eco.switchable is normal and set it to 'ECO' */
+	eco_switchable_str = dstate_getinfo("input.eco.switchable");
+	if (!strcmp(eco_switchable_str, "normal")) {
+		eco_switchable_str = eaton_input_eco_mode_check_range(1);
+	} else {
+		upsdebugx(1, "%s: ECO switch state is: %s , must be normal before switching to ECO", __func__, eco_switchable_str);
+		return eco_mode_auto_off;
+	}
+
+	upsdebugx(1, "%s: ECO Mode was enabled after switching to Bypass Mode", __func__);
+	return eco_mode_auto_on;
+}
+
+/* Function to stop ECO(HE) Mode automatically instead of manually stoping Bypass and then Online Mode */
+static const char *eaton_input_eco_mode_auto_off_fun(double value)
+{
+	const char *bypass_switch_off_str = NULL;
+	const char *eco_switchable_str = NULL;
+
+	NUT_UNUSED_VARIABLE(value);
+
+	/* Check if input.bypass.switch.off is disabled and set it to 'off' */
+	bypass_switch_off_str = dstate_getinfo("input.bypass.switch.off");
+	if (!strcmp(bypass_switch_off_str, "disabled")) {
+		setvar("input.bypass.switch.off", "off");
+	} else {
+		upsdebugx(1, "%s: Bypass switch off state is: %s , must be disabled before switching off", __func__, bypass_switch_off_str);
+		return NULL;
+	}
+
+	/* Check if input.eco.switchable is 'ECO' and set it to normal */
+	eco_switchable_str = dstate_getinfo("input.eco.switchable");
+	if (!strcmp(eco_switchable_str, "ECO")) {
+		setvar("input.eco.switchable", "normal");
+		buzzmode_set("vendor:mge-hid:normal");
+		/* Get the updated value of input.eco.switchable after setting it to "normal */
+        eco_switchable_str = dstate_getinfo("input.eco.switchable");
+	} else {
+		upsdebugx(1, "%s: ECO switch state is: %s , must be ECO before switching to normal", __func__, eco_switchable_str);
+		return NULL;
+	}
+
+	upsdebugx(1, "%s: ECO Mode was disabled after switching from Bypass Mode", __func__);
+	return eco_switchable_str;
+}
+
+/* Conversion back of eaton_input_eco_mode_auto_off_fun() */
+static double eaton_input_eco_mode_auto_off_nuf(const char *value)
+{
+	const char *bypass_switch_off_str = NULL;
+	const char *eco_switchable_str = NULL;
+    double eco_mode_auto_on = 1;
+	double eco_mode_auto_off = 0;
```

Comment:
Just curious: Did compilers complain when you had `return 1;` from a `double` method? Probably a `return 1.0;` would have solved it, but having these named seems more idiomatic to read the code :)